src/statement-list/statement_list_parser.cpp

```diff
@@ -127,77 +147,234 @@ void statement_list_parsert::find_variables(
+      fill_temp_vars(block, decls);
   }
 }
 
 void statement_list_parsert::fill_input_vars(
-  statement_list_parse_treet::functiont &function,
+  statement_list_parse_treet::tia_modulet &module,
   const exprt &input_vars)
 {
   for(exprt entry : input_vars.operands())
   {
-    irep_idt identifier;
+    std::vector<irep_idt> identifiers;
     typet type;
-    for(exprt part : entry.operands())
+    exprt default_value = nil_exprt();
+    for(auto &part : entry.operands())
     {
-      if(part.get(ID_statement_list_type) == ID_statement_list_identifier)
-        identifier = part.get(ID_value);
-      else
+      if(can_cast_expr<symbol_exprt>(part))
+      {
+        symbol_exprt expr = to_symbol_expr(part);
+        identifiers.push_back(expr.get_identifier());
+      }
+      else if(part.get(ID_statement_list_type) == ID_statement_list_type_name)
         type = typet(part.id());
+      else
+        default_value = part;
     }
 
-    statement_list_parse_treet::var_declarationt declaration(identifier, type);
-    function.add_var_input_entry(declaration);
+    for(const auto &identifier : identifiers)
+    {
+      statement_list_parse_treet::var_declarationt declaration{identifier,
+                                                               type};
+      if(default_value.is_not_nil())
+        declaration.default_value = default_value;
+      module.add_var_input_entry(declaration);
+    }
   }
 }
 
 void statement_list_parsert::fill_inout_vars(
-  statement_list_parse_treet::functiont &function,
+  statement_list_parse_treet::tia_modulet &module,
   const exprt &inout_vars)
 {
   for(exprt entry : inout_vars.operands())
   {
-    irep_idt identifier;
+    std::vector<irep_idt> identifiers;
     typet type;
-    for(exprt part : entry.operands())
+    exprt default_value = nil_exprt();
+    for(const auto &part : entry.operands())
     {
-      if(part.get(ID_statement_list_type) == ID_statement_list_identifier)
-        identifier = part.get(ID_value);
-      else
+      if(can_cast_expr<symbol_exprt>(part))
+      {
+        symbol_exprt expr = to_symbol_expr(part);
+        identifiers.push_back(expr.get_identifier());
+      }
+      else if(part.get(ID_statement_list_type) == ID_statement_list_type_name)
         type = typet(part.id());
+      else
+        default_value = part;
     }
 
-    statement_list_parse_treet::var_declarationt declaration(identifier, type);
-    function.add_var_inout_entry(declaration);
+    for(auto &identifier : identifiers)
+    {
+      statement_list_parse_treet::var_declarationt declaration{identifier,
+                                                               type};
+      if(default_value.is_not_nil())
+        declaration.default_value = default_value;
+      module.add_var_inout_entry(declaration);
+    }
   }
 }
 
 void statement_list_parsert::fill_output_vars(
-  statement_list_parse_treet::functiont &function,
+  statement_list_parse_treet::tia_modulet &module,
```

Comment:
A lot of these functions feel very redundant. Perhaps an extract function refactoring could be applied?

Reply:
Will be included in a future PR.